compaction/compaction_manager.cc

```diff
@@ -457,9 +457,7 @@ future<sstables::compaction_result> compaction_task_executor::compact_sstables(s
 future<> compaction_task_executor::update_history(table_state& t, const sstables::compaction_result& res, const sstables::compaction_data& cdata) {
     auto ended_at = std::chrono::duration_cast<std::chrono::milliseconds>(res.stats.ended_at.time_since_epoch());
 
-    if (_cm._sys_ks) {
-        auto sys_ks = _cm._sys_ks; // hold pointer on sys_ks
-
+    if (auto sys_ks = _cm._sys_ks.get_permit()) {
         co_await utils::get_local_injector().inject("update_history_wait", utils::wait_for_message(120s));
```

Comment:
IMO we're introducing a lot of complexity here, and I don't see what we're getting back.

Usually we have destruction in inverse order of construction, with an intermediate drain. Why do we need this thing? Circular dependencies?

Having the system keyspace available conditionally makes the callers more fragile.

Reply:
Added reference to above patches in the git commit message